sale_payment_mgmt/models/account_payment.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2020 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class AccountPayment(models.Model):
+    _inherit = "account.payment"
+
+    def post(self):
+        obj = self
+        if (
+            self.env.user.has_group("sales_team.group_sale_salesman")
```

Comment:
Only first group check is needed to be checked, as the group inheritance assures that the rest of upper groups will have the lower ones.